plugin/pkg/scheduler/factory/factory.go

```diff
@@ -238,6 +246,30 @@ func (f *ConfigFactory) CreateFromKeys(predicateKeys, priorityKeys sets.String,
 	}, nil
 }
 
+func (f *ConfigFactory) getNextPod() *api.Pod {
+	for {
+		pod := f.PodQueue.Pop().(*api.Pod)
+		if f.responsibleForPod(pod) {
+			glog.V(4).Infof("About to try and schedule pod %v", pod.Name)
+			return pod
+		}
+	}
+}
+
+func (f *ConfigFactory) responsibleForPod(pod *api.Pod) bool {
+	if f.SchedulerName == api.DefaultSchedulerName {
+		if pod.Annotations[SchedulerAnnotationKey] == f.SchedulerName || pod.Annotations[SchedulerAnnotationKey] == "" {
+			return true
+		}
+		return false
+	} else {
+		if pod.Annotations[SchedulerAnnotationKey] == f.SchedulerName {
```

Comment:
nit: why do we need the if/else here? just `return pod.Annotations[...] == ...`.  same with the above as well